[GTK] Properly set the path for a newly created WebExtension

ItsJamie9494 opened the pull request:
#### 8c065d25f2e27af2578713a5b2e22225d1dc1316
<pre>
[GTK] Properly set the path for a newly created WebExtension
<a href="https://bugs.webkit.org/show_bug.cgi?id=300588">https://bugs.webkit.org/show_bug.cgi?id=300588</a>

Reviewed by Patrick Griffis.

When fixing the build on Clang, we accidentally removed the
code that sets the resourceBaseURL for the extension.
Somehow the tests never caught that failure. This properly
sets the path and properly checks and adjusts the path as
necessary

* Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGtk.cpp:
(WebKit::WebExtension::WebExtension):

Canonical link: <a href="https://commits.webkit.org/301403@main">https://commits.webkit.org/301403@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/17ec10bd464873e65135730dca8e80408579b91b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125840 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45502 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36253 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132711 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77712 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46185 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54061 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95865 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63976 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128788 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36930 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112527 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76356 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35836 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76182 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106709 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30924 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135392 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52626 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40366 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104332 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53075 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108737 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104060 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26507 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49429 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27750 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/49976 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52520 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58331 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51868 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55216 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53563 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

TingPing commented on Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGtk.cpp:
```diff
@@ -51,6 +51,7 @@ WebExtension::WebExtension(GFile* resourcesFile, RefPtr<API::Error>& outError)
     // and leave only the protocol.
     if (!m_resourceBaseURL.hasPath())
         m_resourceBaseURL.setPath("/"_s);
+    m_resourceBaseURL = URL { String::fromUTF8(baseURL.get()) };
```
You are setting this right after the check above, should that be reversed?

ItsJamie9494 replied:
That check should just be removed in general, the same thing is happening at line 66

mcatanzaro commented on Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGtk.cpp:
```diff
@@ -46,11 +46,7 @@ WebExtension::WebExtension(GFile* resourcesFile, RefPtr<API::Error>& outError)
-    // and leave only the protocol.
-    if (!m_resourceBaseURL.hasPath())
-        m_resourceBaseURL.setPath("/"_s);
+    m_resourceBaseURL = URL { String::fromUTF8(baseURL.get()) };
```
This is really simpler than adding the missing / before creating `m_resourceBaseURL` for the first time?

ItsJamie9494 replied:
We never actually created m_resourceBaseURL here. As well, adding the missing / before is what I had originally done (but using GLib functions that proved to be unsafe). I guess we could create the string before this, including the missing / in that string (possibly via makeString), but that would be a comment on lines 60, not here

ItsJamie9494 committed:
[GTK] Properly set the path for a newly created WebExtension
https://bugs.webkit.org/show_bug.cgi?id=300588

Reviewed by Patrick Griffis.

When fixing the build on Clang, we accidentally removed the
code that sets the resourceBaseURL for the extension.
Somehow the tests never caught that failure. This properly
sets the path and properly checks and adjusts the path as
necessary

* Source/WebKit/UIProcess/Extensions/gtk/WebExtensionGtk.cpp:
(WebKit::WebExtension::WebExtension):

Canonical link: https://commits.webkit.org/301403@main
